src/grammar/execute_immediate.rs

```diff
@@ -0,0 +1,453 @@
+use crate::grammar::{parse_expr, parse_ident};
+use crate::parser::{safe_loop, Parser};
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+use super::parse_into_clause;
+
+pub(crate) fn parse_execute_immediate(p: &mut Parser) {
+    p.start(SyntaxKind::ExecuteImmediateStmt);
+    p.expect(T![execute]);
+    p.expect(T![immediate]);
+    // Parse String
+    if !p.eat(T![quoted_literal]) {
+        parse_ident(p, 1..1);
+    }
+    if p.at(T![into]) {
+        parse_into_clause(p, true);
+    }
+    if p.at(T![bulk]) {
+        parse_bulk_into_clause(p);
+    }
+    if p.at(T![using]) {
+        parse_using_clause(p);
+    }
+    if [T![return], T![returning]].contains(&p.current()) {
+        parse_return_into_clause(p);
+    }
+    p.eat(T![;]);
+    p.finish();
+}
+
+fn parse_using_clause(p: &mut Parser) {
+    p.start(SyntaxKind::UsingClause);
+    p.expect(T![using]);
+    safe_loop!(p, {
+        if [T![in], T![out]].contains(&p.current()) {
+            p.eat(T![in]);
+            p.eat(T![out]);
+        }
```

Comment:
Is the `.contains` needed? Especially since `p.eat` doesn't touch our tokens if those wouldn't be present.
```suggestion
        p.eat(T![in]);
        p.eat(T![out]);
```